src/aiida/cmdline/groups/dynamic.py

```diff
@@ -179,11 +178,8 @@ def list_options(self, entry_point: str) -> list:
             # we simply remove all ``NoneType`` and the remaining type should be the type of the option.
             if hasattr(field_info.annotation, '__args__'):
                 args = list(filter(lambda e: e != type(None), field_info.annotation.__args__))
-                if len(args) > 1:
```

Comment:
Does this have any implications in practical usage? If click supports only one type, will it reject if the second type is provided?

Reply:
It is a limitation of the design to use the pydantic model to automatically try and convert that into a click interface. The pydantic models allow specifying multiple types for a field, e.g.
```python
class Model(BaseModel):
    filepath_files: str | pathlib.Path
```
But click, as far as I can tell, does not support this for their `click.option` interface. Trying to do
```python
import click

@click.command()
@click.option('--filepath-files', type=(str, pathlib.Path)
def test(filepath_files):
    pass
```
will result in an option that expects _two_ values; the first one will be cast to `str` and the second to a `pathlib.Path`.

So I am not sure how to approach this in a better way. I think I am forced to just pick one type if the model field defines multiple. This is why I originally had the `UserWarning` since just discarding the additional types could be unexpected to the user. But we see the downsides, and the "user" here is really us. The `DynamicEntryPointGroup` _is_ public technically, but I doubt anyone is using it and we also don't advertise it as such. So as long as it works for our purposes I think it should be fine for now.

Reply:
Thanks. I was mainly concerned about any impact on actual CLI (verdi) usage.